dev/com.ibm.ws.kernel.service/src/com/ibm/ws/kernel/service/location/internal/WsLocationAdminImpl.java

```diff
@@ -335,7 +340,36 @@ protected WsLocationAdminImpl(Map<String, Object> config) {
 
         addResourcePath(bootstrapLib.getNormalizedPath());
 
-        SymbolRegistry.getRegistry().addStringSymbol(WsLocationConstants.LOC_VARIABLE_SOURCE_DIRS, (String) config.get(WsLocationConstants.LOC_VARIABLE_SOURCE_DIRS));
+        String variableSourceDirs = (String) config.get(WsLocationConstants.LOC_VARIABLE_SOURCE_DIRS);
+        SymbolRegistry.getRegistry().addStringSymbol(WsLocationConstants.LOC_VARIABLE_SOURCE_DIRS, variableSourceDirs);
+
+        bundleContext.registerService(CheckpointHookFactory.class, (p) -> new CheckpointHook() {
+            @Override
+            public void restore() {
+                String newVariableSourceDirs = getEnv(BootstrapConstants.ENV_VARIABLE_SOURCE_DIRS);
+                if (!newVariableSourceDirs.equals(variableSourceDirs)) {
+                    SymbolRegistry.getRegistry().replaceStringSymbol(WsLocationConstants.LOC_VARIABLE_SOURCE_DIRS, newVariableSourceDirs);
+                }
+            };
+        }, null);
+    }
+
+    String getEnv(final String key) {
+        if (key == null)
+            return null;
+
+        String value = null;
+        try {
+            value = AccessController.doPrivileged(new java.security.PrivilegedExceptionAction<String>() {
+                @Override
+                public String run() throws Exception {
+                    return System.getenv(key);
+                }
+            });
+        } catch (Exception ex) {
+
+        }
+        return value;
```

Comment:
I don't see the need to use a PrivilegedExceptionAction here and with Java 8 you can reduce this to:
```java
        return AccessController.doPrivileged((PrivilegedAction<String>)() -> {
            return System.getenv(key);
        });
```

Reply:
done